fabric-models-v0/src/testmodClient/java/net/fabricmc/fabric/test/model/SpecificModelReloadListener.java

```diff
@@ -16,23 +16,15 @@
 
 package net.fabricmc.fabric.test.model;
 
-import java.util.Arrays;
-import java.util.Collection;
-
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.render.model.BakedModel;
 import net.minecraft.resource.ResourceManager;
-import net.minecraft.resource.SinglePreparationResourceReloader;
+import net.minecraft.resource.SynchronousResourceReloader;
 import net.minecraft.util.Identifier;
-import net.minecraft.util.Unit;
-import net.minecraft.util.profiler.Profiler;
 
 import net.fabricmc.fabric.api.client.model.BakedModelManagerHelper;
-import net.fabricmc.fabric.api.resource.IdentifiableResourceReloadListener;
-import net.fabricmc.fabric.api.resource.ResourceReloadListenerKeys;
 
-public class SpecificModelReloadListener extends SinglePreparationResourceReloader<Unit> implements IdentifiableResourceReloadListener {
-	public static final SpecificModelReloadListener INSTANCE = new SpecificModelReloadListener();
+public class SpecificModelReloadListener implements SynchronousResourceReloader {
```

Comment:
`getSpecificModel` needs a null-check on the model. This will make the error happen at load time rather than at render time.